src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -904,7 +903,16 @@ protected QuestionDef parseUpload(IFormElement parent, Element e, int controlUpl
     }
 
     protected QuestionDef parseControl(IFormElement parent, Element e, int controlType) {

```

Comment:
this is an overloaded method for another `parseControl` method and should not contain any logic except calling the other `parseControl` method